### PR TITLE
Add Conv2D and MaxPool reference

### PR DESCRIPTION
**Pull request by apertovs**

### CODE REVIEW

**Review by l-bat — APPROVED** [review]

File: tests/unit/kernels_reference.cpp [hunk 1]
```diff
@@ -174,3 +175,81 @@ std::vector<float> dw::reference::Transpose(const float* in, size_t rows, size_t
 
     return std::move(inT);
 }
+
+void dw::reference::CPUMaxPooling2DForward(const float* input, const deepworks::Shape& shape, float* output, const std::array<int, 2>& kernel,
```
Comment (TolyaTalamanov) [inline tests/unit/kernels_reference.cpp:179]: Why not a `Tensor` ?

Comment (apertovs) [reply tests/unit/kernels_reference.cpp:179]: Why should there be a tensor? I think all reference implementations should have the same API.

Comment (TolyaTalamanov) [reply tests/unit/kernels_reference.cpp:179]: New ref implementation take Tensor instead of raw data, we used raw cuz tensor wasn't implemented

Comment (TolyaTalamanov) [reply tests/unit/kernels_reference.cpp:179]: @l-bat you will use it, your turn

File: tests/unit/kernels_reference.cpp [hunk 2]
```diff
@@ -174,3 +175,78 @@ std::vector<float> dw::reference::Transpose(const float* in, size_t rows, size_t
...
+
+    const torch::Tensor input_tensor = torch::from_blob(const_cast<void*>(reinterpret_cast<const void*>(input)), input_shape, options);
+    const torch::Tensor weights_tensor = torch::from_blob(const_cast<void*>(reinterpret_cast<const void*>(weights)), weights_shape, options);
+    const torch::Tensor bias_tensor = torch::from_blob(const_cast<void*>(reinterpret_cast<const void*>(bias)), bias_shape, options);
+
+    torch::Tensor result_tensor = torch::conv2d(input_tensor, weights_tensor, bias_tensor, {stride[0], stride[1]}, {padding[0], padding[1]});
+    std::copy_n(result_tensor.data_ptr<float>(), result_tensor.numel(), output);
+}
+
+void dw::reference::CPUConvolution2DBackward(const float* input, const float* grad_output, const deepworks::Shape& shape,
```
Comment (l-bat) [inline tests/unit/kernels_reference.cpp:210]: `shape` it's hard to understand  what shape I should pass to the function

Comment (TolyaTalamanov) [reply tests/unit/kernels_reference.cpp:210]: because the API with raw pointer isn't descriptive at all

**Comment by TolyaTalamanov** [comment]
Great :fire: 
Thanks! :+1:

**Pull request merged**